clang/lib/AST/ExprConstant.cpp

```diff
@@ -4426,7 +4431,7 @@ static CompleteObject findCompleteObject(EvalInfo &Info, const Expr *E,
 
     // Unless we're looking at a local variable or argument in a constexpr call,
     // the variable we're reading must be const.
-    if (!Frame) {
+    if (AK != clang::AK_ReferenceInitialization && !Frame) {
```

Comment:
Can you explain why you are pairing these checks? I am missing the logic, maybe a comment should be added?